File: frontend/src/models/product/productFilterModel.ts
import { ProductListModel } from "@/models/product/productListModel";
import { SortDirection } from "@/models/sortDirection";

export interface ProductFilterModel {
    searchString?: string;
    orderByProperty?: keyof ProductListModel,
    sortDirection?: SortDirection;
    categories: string[];
    minAmount?: number;
    maxAmount?: number;
}

File: HomeStorage.Logic/Models/ProductModels/ProductFilterModel.cs
using System.ComponentModel;

namespace HomeStorage.Logic.Models.ProductModels
{
    public class ProductFilterModel
    {
        public string? OrderByProperty { get; set; }
        public ListSortDirection SortDirection { get; set; } = ListSortDirection.Ascending;
        public List<Guid> Categories { get; set; } = [];
        public double? MaxAmount { get; set; }
        public double? MinAmount { get; set; }
        public string? SearchString { get; set; }
    }
}
